index.js

```diff
@@ -128,14 +144,17 @@ class Ky {
 		this._options.prefixUrl = String(this._options.prefixUrl || '');
 		this._input = String(input || '');
 
-		if (this._options.prefixUrl && this._input.startsWith('/')) {
-			throw new Error('`input` must not begin with a slash when using `prefixUrl`');
-		}
-		if (this._options.prefixUrl && !this._options.prefixUrl.endsWith('/')) {
-			this._options.prefixUrl += '/';
+		const isFullUrlInput = isAbsoluteUrl(this._input);
+		const isFullUrlPrefix = isAbsoluteUrl(this._options.prefixUrl);
+		const documentBaseUri = _globalThis.document && _globalThis.document.baseURI;
+
+		if (!isFullUrlInput && !isFullUrlPrefix && !documentBaseUri) {
```

Comment:
I think we have to check all three here.

- If `input` is a full URL (i.e., `https://foo.com`), we don't need a prefix and we don't need `document.baseURI`.
- If `input` is relative and `prefixUrl` is a full URL (i.e., `https://foo.com`), we also don't need `document.baseURI`.
- If neither `input` nor `prefixUrl` are full URLs, then we need `document.baseURI` to be present. If it's not present, we throw an error.

Here I was wondering how to handle the case where a user passes full URLs for both `input` and `prefixUrl`. I was considering throwing an error in this case, but then saw the example of using `input` to set a query parameter in a URL.

Reply:
Maybe there should be an option to pass custom `baseURI`?

Reply:
> If input is absolute, it should take priority over prefixUrl

I mentioned this briefly in #11. I tried to characterize it as _partially implemented_ URL resolution, which I find to be somewhat confusing behavior. It means the option neither strictly behaves like a prefix, nor like any other link or base URL on the web platform. But of course, it can be documented as such. I just like the simplicity of the prefix behavior. It's easy to reason about.

> I was considering throwing an error in this case, but then saw the example of using input to set a query parameter in a URL.

I'd be fine with that, personally. My intention with those tests that have `prefixUrl` ending in a query string was to make it obvious that `ky` isn't just naively joining the URLs with a `/`. I doubt many people are actually using a query string prefix. The point is that semantically, they could, if they needed to for some reason. This would be a breaking change, at least in edge cases like that.

A bigger issue, in my mind is that this feels surprising:

```js
ky('http://foo.com', {prefixUrl: '/api'});  // prefixUrl is silently ignored
ky('http://foo.com', {prefixUrl: 'http://bar.com'});  // Error is thrown
```

I don't think we should do that. I want it to either throw in both cases, or in neither.